### DISCUSSION TITLE
Spell Checker not working...

### DISCUSSION THREAD

**Discussion by shary998** [florisboard/florisboard#1848] [Q&A]
I'm trying to use spell checker for Spanish language but it is not working.
Predictions are taking the index of predicted word with itself like(google0,google1,google3).. How to solve this issue?

**Comment by magnus-ISU** [comment]
Downgrade to the previous version (for example, on F-droid. Unsure if google store supports this).

I was initially pretty mad about this too, so didn't upgrade. But I had some crashes and performance issues which the current rewrite seems to fix, so I eventually upgraded. I hope we get support soon too, as losing features from an update didn't sit well with me either.

**Comment by tsiflimagas** [comment]
If you look through issues or release notes you'll know why it is like that. There had been an attempt to implement a word suggestions mechanism in Kotlin, which proved to be a wrong decision as there's the constraint of the JVM heap memory, which was the cause of Out Of Memory conditions, performance issues etc. Since it was the reason for a storm of reports and a new C++ spelling/suggestions mechanism was decided to be implemented, that old mechanism was removed. The new mechanism is the next big thing to come to FlorisBoard, and set to 0.4.0 milestone. Right now, there's some preparatory work being done (and some miscellaneous changes I guess).
